kube-runtime/src/wait.rs

```diff
@@ -0,0 +1,95 @@
+use futures::TryStreamExt;
+use kube::{Api, Resource};
+use serde::de::DeserializeOwned;
+use snafu::{futures::TryStreamExt as _, Snafu};
+use std::fmt::Debug;
+
+use crate::watcher::{self, watch_object};
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("failed to probe for whether the condition is fulfilled yet: {}", source))]
+    ProbeFailed {
+        #[snafu(backtrace)]
+        source: watcher::Error,
+    },
+}
```

Comment:
Is it still worth keeping this around around when there is only one possible cause?